prediction_market_agent/agents/microchain_agent/memory_functions.py

```
@@ -94,6 +97,44 @@ def __call__(self, context: str) -> str:
+    as part of the prompt history of each agent.
+    """
+    entries_from_latest_run: dict[str, list[LongTermMemories]] = {}
+    for agent_id in agent_identifiers:
+        ltm = LongTermMemoryTableHandler.from_agent_identifier(agent_id)
+        entries_for_agent = ltm.search()
+        # We need the 2nd GameRoundEnd. If there is only one, then start_date should be None.
+        game_round_occurrences = [
+            i
+            for i in entries_for_agent
+            if i.metadata_dict is not None and "content" in i.metadata_dict
+            and f"{GameRoundEnd.__name__}(" in i.metadata_dict["content"]
```

Ruff (0.8.2)

Undefined name `GameRoundEnd`

(F821)